src/app/app.component.html

```diff
@@ -18,20 +18,25 @@
 
 <div class="margin-row">
 	<div class="container">
-		<form #addPackageNameForm onsubmit="return false">
-			<mat-form-field appearance="outline">
-				<mat-label>npm package name</mat-label>
-				<input matInput [formControl]="addPackage" [matAutocomplete]="auto" />
-				<mat-hint>Ex: 'canjs'</mat-hint>
-				<mat-error appErrorHandler [errors$]="addPackage.errors | toObservable" [handleErrors]="packageErrorsHandler">
-				</mat-error>
-
-				<mat-autocomplete #auto="matAutocomplete">
-					<mat-option *ngFor="let option of autocompleteOptions$ | async" [value]="option"> {{ option }} </mat-option>
-				</mat-autocomplete>
-			</mat-form-field>
-
-			<button mat-raised-button color="primary" [disabled]="!addPackage.value || addPackage.invalid">Add</button>
+		<!-- auto complete form -->
+		<form onsubmit="return false">
+			<!-- auto complete -->
+			<app-autocomplete
+				[autocomplateOptions]="autocompleteOptions$ | async"
```

Comment:
how do we feel about naming the property `autocomplateOptions$`?

Reply:
I don't feel good about that. I rather avoid passing the reference to the observable like `@Input() autocomplateOptions$: Observable<string[]>` , because it is harder to work with an observable than just an array of values